File: 509-fibonacci-number/509-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        int x=0;
        int y=1;
        int result=x+y;    
        int array[n+1];
        array[0]=x;
    
        for(int i=1;i<n+1;i++){
            array[i]=result;
            result=x+y;
            x=y;
            y=result;
        }
        return array[n];
    }
};

File: 509-fibonacci-number/509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        x = 0
        y = 1
        result = x+y
        l=[]
        l.append(x)
        for i in range(n):
            l.append(result)
            result = x+y
            x = y
            y = result

        return l[n]